### PR TITLE
[debugger plugin] tensor-widget: Implement some basic features

### PR DESCRIPTION
**Pull request by caisq**
* Motivation for features / changes
  * Continue implementation of TensorWidget.

* Technical description of changes
  - Add a demo directory, which builds and launches a demo based on concrete TensorFlow.js tensors.
  - Add demo/package.json to facilitate pulling in dependencies of the demo
  - Add package.json to facilitate unit testing and linting
  - Add tslint to tensor_widget
  - Add karma unit testing to tensor_widget
  - Add helper modules for dtype, shape and numeric values
  - Add basic logic for rendering health pills
  - Add basic logic for rendering the header portion of the image 
  - Add unit test for the aforementioned modules and logic
  - Add developer workflow info to README.md
  - Note that the actual text value in the tabular view will be implemented in a later PR.

* Screenshots of UI changes
image

* Detailed steps to verify changes work correctly (as executed by you)
  - See the Developer Workflow section in README.md

* Alternate designs / implementations considered

* Review requirements
  - WANT_LGTM from nfelt@ or stephanwlee@ about code structure and the code outside tensorboard/components/tensor_widget
  - WATN_LGTM from davidsoergel@ or bilsechi@ about the implementation, test and demo code in tensorboard/components/tensor_widget

### CODE REVIEW

**Comment by nfelt** [comment]
@caisq do you mind providing some guidance as to your intent with setting 4 reviewers?  I.e. is this WANT_LGTM=any, WANT_LGTM=all, or do you want specific parts of the PR reviewed by different people?  Just trying to make sure we're on the same page.

**Comment by caisq** [comment]
@nfelt Good question. I  added two WANT_LGTM items to the description of this PR. It's slightly more complex than =any or =all. Let me know if that is okay.

File: tensorboard/components/tensor_widget/numeric-helper-test.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    decimalPlaces = 0;
+    expect(numericValueToString(x, decimalPlaces, 'exponential'))
+        .toEqual('0e+0');
+    decimalPlaces = 2;
+    expect(numericValueToString(x, decimalPlaces, 'exponential'))
+        .toEqual('0.00e+0');
+  });
+
+  it('Large positive value', () => {
```
Comment (stephanwlee) [inline tensorboard/components/tensor_widget/numeric-helper-test.ts]: Comments about naming test in general: mocha, especially when using the `it` syntax, is supposed to be BDD-style. Please describe expected behavior, not just short description like `zero` or `large negative value` (also consistency in capitalization would be nice).

File: tensorboard/components/tensor_widget/dtype-helper-test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {isIntegerDType, isFloatDType} from './dtype-helper';
+
+describe('isIntegerDType', () => {
+  it('unsigned ints', () => {
+    expect(isIntegerDType('uint8')).toEqual(true);
```
Comment (stephanwlee) [inline tensorboard/components/tensor_widget/dtype-helper-test.ts]: chai is supposed to be fluent. Please prefer https://www.chaijs.com/api/bdd/#true

Comment (stephanwlee) [reply tensorboard/components/tensor_widget/dtype-helper-test.ts]: Separately, `to.equal` is the pattern we use.

**Comment by caisq** [comment]
As discussed offline with @stephanwlee , I'll break this PR into two smaller ones, after the infrastructure change in https://github.com/tensorflow/tensorboard/pull/2472 is checked in. So I'll close this PR for now.

**Pull request closed without merging**